packages/grid/_modules_/grid/hooks/root/useEvents.ts

```diff
@@ -199,8 +199,8 @@ export function useEvents(gridRootRef: React.RefObject<HTMLDivElement>, apiRef:
       return () => {
         logger.debug('Clearing all events listeners');
         api.publishEvent(GRID_UNMOUNT);
-        gridRootElem.removeEventListener(GRID_CLICK, onClickHandler, { capture: true });
-        gridRootElem.removeEventListener(GRID_MOUSE_HOVER, onHoverHandler, { capture: true });
+        gridRootElem.removeEventListener(GRID_CLICK, onClickHandler);
```

Comment:
the event bubble up so if you attach them to the root then they won't fire when you click an element within it.

Reply:
I don't understand the reasoning. 

Regarding the outcome, the demos in https://deploy-preview-1156--material-ui-x.netlify.app/components/data-grid/selection/#data-grid-selection seems to work correctly and the tests are green.

Reply:
checkout #1158

Reply:
Before we move forward with #1158, could you report one bug with the current changes?

Reply:
> the event bubble up so if you attach them to the root then they won't fire when you click an element within it.

A counterexample that demonstrates this affirmation is false: https://codesandbox.io/s/nifty-bush-brmqw?file=/src/App.tsx

Reply:
Yes I agree I got really confused here with the bubbling. Just need to check the params are still ok but I think they should be

Reply:
Now, I personally think that #1158 would be even better than #1156. It removes the capture phase, which should be more intuitive for developers **and** integrate with the synthetic event system of React, perfect for using custom logic 👌

Reply:
The nice thing about #1158 is that it could use the helper to generate the params...
The bad thing is atm we generate a new function at every render so react.memo is not possible

Reply:
Ok, what do you recommend as the best next step?

Reply:
I will refactor a bit more and merge #1158 with the comments above